sulong/projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/LLVMParser.java

```diff
@@ -152,17 +155,28 @@ private void defineAlias(String aliasName, boolean isAliasExported, SymbolImpl v
             defineAlias(global.getName(), aliasName, isAliasExported);
         } else if (value instanceof GlobalAlias) {
             GlobalAlias target = (GlobalAlias) value;
-            defineAlias(target);
+            defineAlias(target, targetDataLayout);
             defineAlias(target.getName(), aliasName, isAliasExported);
         } else if (value instanceof CastConstant) {
             // TODO (chaeubl): this is not perfectly accurate as we are loosing the type cast
             CastConstant cast = (CastConstant) value;
-            defineAlias(aliasName, isAliasExported, cast.getValue());
+            defineAlias(aliasName, isAliasExported, cast.getValue(), targetDataLayout);
+        } else if (value instanceof GetElementPointerConstant) {
+            GetElementPointerConstant elementPointerConstant = (GetElementPointerConstant) value;
+            defineExpressionSymbol(aliasName, isAliasExported, elementPointerConstant, targetDataLayout);
         } else {
             throw new LLVMLinkerException("Unknown alias type: " + value.getClass());
         }
     }
 
+    private void defineExpressionSymbol(String aliasName, boolean isAliasExported, GetElementPointerConstant elementPointerConstant, DataLayout targetDataLayout) {
+        LLVMSymbol baseSymbol = runtime.getFileScope().get(elementPointerConstant.getBasePointer().toString());
+        LLVMGetElementPtrNode elementPtrNode = (LLVMGetElementPtrNode) elementPointerConstant.createNode(runtime, targetDataLayout, GetStackSpaceFactory.createAllocaFactory());
```

Comment:
(cont from other comment)

... so here, instead of creating the node, I'd create a factory object or a lambda that captures everything it needs and is able to create multiple GEP nodes on the fly.